### PR TITLE
Bytecode instruction with int operand

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java [hunk 1]
```diff
@@ -110,10 +110,12 @@ public MethodVisitor visitMethod(int access, String name, String desc, String si
       instructions = new ArrayList<>();
       successors = new ArrayList<>();
     }
-
     void addInsn(int opcode) {
       instructions.add(new Instruction(opcode));
     }
+    void addInsn(int opcode, int operand) {
```
Comment (saberduck) [inline java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java]: formatting?

File: java-frontend/src/test/java/org/sonar/java/bytecode/se/BytecodeEGWalkerExecuteTest.java [hunk 2]
```diff
@@ -64,6 +64,23 @@ public void test_areturn() throws Exception {
...
+  @Test
+  public void test_load() throws Exception {
+    int[] loadRefOpcodes = new int[] {Opcodes.ILOAD, Opcodes.LLOAD, Opcodes.FLOAD, Opcodes.DLOAD, Opcodes.ALOAD};
+    for (int loadRefOpcode : loadRefOpcodes) {
+      SymbolicValue loadRef = new SymbolicValue();
+      ProgramState programState = execute(new Instruction(loadRefOpcode, 0), ProgramState.EMPTY_STATE.put(0, loadRef));
+      assertThat(programState.peekValue()).isEqualTo(loadRef);
+      try {
+        // no SV indexed should failed
+        programState = execute(new Instruction(loadRefOpcode, 0), ProgramState.EMPTY_STATE);
```
Comment (saberduck) [inline java-frontend/src/test/java/org/sonar/java/bytecode/se/BytecodeEGWalkerExecuteTest.java]: perhaps bit nicer with `assertThatThrownBy`

File: java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java [hunk 3]
```diff
@@ -155,7 +163,45 @@ void executeInstruction(BytecodeCFGBuilder.Instruction instruction) {
   }
 
   private void handleBlockExit(ProgramPoint programPosition) {
-    programPosition.block.successors().forEach(b -> enqueue(new ProgramPoint(b), programState));
+    BytecodeCFGBuilder.Block block = (BytecodeCFGBuilder.Block) programPosition.block;
+    BytecodeCFGBuilder.Instruction terminator = block.terminator();
+    if (terminator != null) {
+      switch (terminator.opcode) {
```
Comment (SonarTech) [inline java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java]: CRITICAL Add a default case to this switch. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![CRITICAL][CRITICAL] 1 critical

Watch the comments in this conversation to review them.

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'

**Pull request merged**